### PR TITLE
Use the application name in the cache name

### PR DESCRIPTION
**Pull request by zhang-ca**
Default unique per-application Session cache names build with the application context root for JCache session persistence.

Enable option to use additional application name in the session cache names.

### CODE REVIEW

**Comment by zhang-ca** [comment]
#build

**Comment by zhang-ca** [comment]
fixes #24585

**Comment by zhang-ca** [comment]
for #24585

File: dev/com.ibm.ws.session.cache/resources/OSGI-INF/l10n/metatype.properties [hunk 1]
```diff
@@ -29,6 +29,9 @@ uri.desc=Vendor-specific JCache configuration URI, which is passed to the JCache
 cacheSeparator=Cache name separator
 cacheSeparator.desc=The single character used to separate the session meta cache name. The default value should usually be used. 
 
+appInCacheName=Use the application name in the cache names
+appInCacheName.desc=Default unique per-application cache names build with the application context root. Set this property to "true" to use additional application name in the cache names.
```
Comment (jvanhill) [inline dev/com.ibm.ws.session.cache/resources/OSGI-INF/l10n/metatype.properties]: Maybe something more specific.

```By default, per-application JCache session cache names are generated using the context root. When distributing the JCache seeion caches across multiple servers, it is possible to have multiple applications with the same context root that should not share a session cache. Enabling this option will result in the JCache cache names including the application name to help avoid conflicting JCache cache names. The default is "false".```

**Review by jvanhill — CHANGES_REQUESTED** [review]
I left a recommendation in the metatype description for appInCacheName.

I also do not see a unit test that tests this behavior.

File: dev/com.ibm.ws.session.cache/resources/OSGI-INF/l10n/metatype.properties (hunk 1 above)
Comment (zhang-ca) [reply dev/com.ibm.ws.session.cache/resources/OSGI-INF/l10n/metatype.properties]: Thanks, will update the description.

**Comment by posmith** [comment]
L2 message review not required

**Comment by zhang-ca** [comment]
#build

**Comment by zhang-ca** [comment]
#build

**Comment by zhang-ca** [comment]
#build

**Comment by zhang-ca** [comment]
#run-libby-bot

**Comment by zhang-ca** [comment]
#build

File: dev/com.ibm.ws.session.cache/resources/OSGI-INF/l10n/metatype.properties [hunk 2]
```diff
@@ -29,6 +29,9 @@ uri.desc=Vendor-specific JCache configuration URI, which is passed to the JCache
 cacheSeparator=Cache name separator
 cacheSeparator.desc=The single character used to separate the session meta cache name. The default value should usually be used. 
 
+appInCacheName=Use the application name in the cache names
+appInCacheName.desc=By default, per-application JCache session cache names are generated using the context root. When distributing the JCache session caches across multiple servers, it is possible to have multiple applications with the same context root that should not share a session cache. Enabling this option will result in the JCache cache names including the application name to help avoid conflicting JCache cache names. The default is "false".
```
Comment (dmuelle) [inline dev/com.ibm.ws.session.cache/resources/OSGI-INF/l10n/metatype.properties]:
```suggestion
appInCacheName.desc=By default, per-application JCache session cache names are generated by using the context root. When the JCache session caches are distributed across multiple servers, multiple applications with the same context root might exist that must not share a session cache. When this option is enabled, application names are included in JCache cache names to help avoid conflicting JCache cache names. The default value is false. 
```

**Review by dmuelle — CHANGES_REQUESTED** [review]
ID review- let me know if you have any questions, thanks.

File: dev/com.ibm.ws.session.cache/resources/OSGI-INF/l10n/metatype.properties (hunk 2 above)
Comment (zhang-ca) [reply dev/com.ibm.ws.session.cache/resources/OSGI-INF/l10n/metatype.properties]: Thanks Dave for the ID review and update, I will make the changes.

**Review by jvanhill — DISMISSED** [review]

**Comment by zhang-ca** [comment]
#build

**Comment by zhang-ca** [comment]
#run-libby-bot

**Review by dmuelle — APPROVED** [review]
LGTM thanks

**Pull request merged**